tutorial/stream-client.md

```diff
@@ -0,0 +1,261 @@
+        auth = request.authorization
+        if not (auth and check_auth(auth.username, auth.password)):
+            return (
+                "Unauthorized",
+                401,
+                {"WWW-Authenticate": 'Basic realm="Login Required"'},
+            )
+        return f(**kwargs)
+
+    return wrapped_view
+
+
+@app.route("/1.0/sessions/<session_id>/join", methods=["POST"])
+@login_required
+def api_1_0_sessions_join_post(session_id):
+    if not gateway_enabled:
+        return render_error_response("no gateway connected", 503)
+
+    req = {"disconnect_clients": True}
+    resp = gateway.join_session(session_id, req)
+
+    if "status_code" not in resp or resp["status_code"] != 200:
+        return render_error_response("failed to join session", 500)
+
+    return jsonify(resp), 200
+
+
+@app.route("/<session_id>")
+@login_required
+def stream_session(session_id=None):
+    return render_template("stream.html", session_id=session_id)
+
+
+@app.route("/")
+@login_required
+def index():
+    return "Welcome to the Anbox Cloud Stream SDK tutorial!"
+```
+
+Create a `stream.html` file inside `stream-client/static` with the following code:
+
+```html
+<!doctype html>
+<html>
+  <head>
+    <title>Anbox Streaming SDK - Tutorial</title>
+    <script type="module" src="static/anbox-stream-sdk.js"></script>
+    <style>
+      html,
+      body,
+      #anbox-stream {
+        height: 100%;
+        margin: 0;
+        width: 100%;
+      }
+    </style>
+  </head>
+
+  <body>
+    <div id="anbox-stream"></div>
+    <script>
+      const sessionId = "{{ session_id }}";
+      class Connector {
+        async connect() {
+          var url = `${window.location.origin}/1.0/sessions/${sessionId}/join`;
+          const resp = await fetch(url, { method: "POST" });
+          if (resp === undefined || resp.status !== 200)
+            throw new Error("failed to join session");
+
+          const response = await resp.json();
+          if (response === undefined || response.status !== "success")
+            throw new Error(response.error);
+
+          return {
+            id: response.metadata.id,
+            websocket: response.metadata.url,
+            stunServers: response.metadata.stun_servers,
+          };
+        }
+
+        disconnect() {}
+      }
+      var stream;
+      window.onload = () => {
+        const connector = new Connector();
+        stream = new AnboxStream({
+          connector: connector,
+          targetElement: "anbox-stream",
+          enableStats: false,
+          stream: {
+            audio: true,
+            video: true,
+          },
+        });
+        stream.connect();
+      };
+    </script>
+  </body>
+</html>
+```
+
+Next, let's make a copy of the Anbox Stream Gateway server to allow the service to validate it:
+
+    # We use cat here to ensure gateway.crt has the users permissions
+    sudo cat /var/snap/anbox-cloud-appliance/common/gateway/server.crt > gateway.crt
+
```

Comment:
Should users do this in general, or should it be limited to small scale tests?
I feel like this shouldn't be done in production, but I'm not too sure if it's a problem.
@adglkh @morphis @jat-canonical Wdyt?

Reply:
@ajanon what specifically do you mean?

Reply:
No, it's not about using the same certificate (we would still miss the private key for that). This is simply to make the flask service add the cert to its list of trusted certs so it can validate the connection with the gateway service at the TLS level. Does that answer your question?